src/account/operations/output_claiming.rs

```diff
@@ -231,6 +231,24 @@ impl AccountHandle {
             let mut new_native_tokens = NativeTokensBuilder::new();
             // check native tokens
             for output_data in outputs {
+                if let Some(native_tokens) = output_data.output.native_tokens() {
+                    // Skip output if the max native tokens count would be exceeded
+                    if new_native_tokens.len() + native_tokens.len() > NativeTokens::COUNT_MAX.into() {
+                        log::debug!("[OUTPUT_CLAIMING] skipping output to not exceed the max native tokens count");
+                        continue;
+                    }
+                    new_native_tokens.add_native_tokens(native_tokens.clone())?;
```

Comment:
A side-node: we could avoid that kind of clones by taking ownership of `outputs_to_claim` since it's not used anymore after. But it seems like the std `chunks` doesn't work with ownership and we would have to use the `itertools` crate. Not a terribly bad idea if we can remove some nested clones but maybe something that can be done later and not in this PR.

Reply:
Already thought about using itertools at another time for something, but didn't want to add it as dependency just because of that, if we have more things like this it probably makes sense